### PR TITLE
Upgrade to SDK 0.13.39

### PR DESCRIPTION
**Pull request by dasormeter**
Upgrade daml-on-x example to latest released SDK 0.13.39 (aka 100.13.39)
Details of changes in CHANGELOG.md

Most impactful changes are move of allocateParty interface to async SubmissionResult response and change in envelope for KV submission and consumption

### CODE REVIEW

File: src/main/scala/com/daml/ledger/damlonxexample/ExampleInMemoryParticipantState.scala [hunk 1]
```diff
@@ -361,10 +355,12 @@ class ExampleInMemoryParticipantState(
...
           .get(entryId.getEntryId)
           .map { blob =>
-            KeyValueConsumption.logEntryToUpdate(
-              entryId,
-              KeyValueConsumption.unpackDamlLogEntry(blob)
-            )
+            val logEntry = Envelope.open(blob) match {
+              case Left(err)                                 => sys.error(s"getUpdate: cannot open envelope: $err")
+              case Right(Envelope.LogEntryMessage(logEntry)) => logEntry
+              case Right(_)                                  => sys.error(s"getUpdate: Envelope did not contain log entry")
```
Comment (miklos-da) [inline src/main/scala/com/daml/ledger/damlonxexample/ExampleInMemoryParticipantState.scala]: No string interpolation required here.

File: src/main/scala/com/daml/ledger/damlonxexample/ExampleInMemoryParticipantState.scala [hunk 2]
```diff
@@ -512,15 +505,36 @@ class ExampleInMemoryParticipantState(
...
+      state.store
+        .getOrElse(
+          entryId.getEntryId,
+          sys.error(s"getLogEntry: Cannot find ${Pretty.prettyEntryId(entryId)}!")
+        )
+    ) match {
+      case Right(Envelope.LogEntryMessage(logEntry)) =>
+        logEntry
+      case _ =>
+        sys.error(s"getLogEntry: Envelope did not contain log entry")
```
Comment (miklos-da) [inline src/main/scala/com/daml/ledger/damlonxexample/ExampleInMemoryParticipantState.scala]: No string interpolation required here.

File: src/main/scala/com/daml/ledger/damlonxexample/ExampleInMemoryParticipantState.scala [hunk 3]
```diff
@@ -512,15 +505,36 @@ class ExampleInMemoryParticipantState(
...
+
+  private def getDamlState(state: State, key: Proto.DamlStateKey): Option[Proto.DamlStateValue] =
     state.store
       .get(NS_DAML_STATE.concat(KeyValueCommitting.packDamlStateKey(key)))
-      .map(DamlStateValue.parseFrom)
+      .map(
+        blob =>
+          Envelope.open(blob) match {
+            case Right(Envelope.StateValueMessage(v)) => v
+            case _                                    => sys.error(s"getDamlState: Envelope did not contain a state value")
```
Comment (miklos-da) [inline src/main/scala/com/daml/ledger/damlonxexample/ExampleInMemoryParticipantState.scala]: No string interpolation required here.

**Comment by dasormeter** [comment]
> Please also add a note about new mandatory implementation of the currentHealth method of the ReportsHealth trait.

Added to changelog

File: src/main/scala/com/daml/ledger/damlonxexample/ExampleInMemoryParticipantState.scala (hunk 1 above)
Comment (dasormeter) [reply src/main/scala/com/daml/ledger/damlonxexample/ExampleInMemoryParticipantState.scala]: addressed

File: src/main/scala/com/daml/ledger/damlonxexample/ExampleInMemoryParticipantState.scala (hunk 3 above)
Comment (dasormeter) [reply src/main/scala/com/daml/ledger/damlonxexample/ExampleInMemoryParticipantState.scala]: addressed

File: src/main/scala/com/daml/ledger/damlonxexample/ExampleInMemoryParticipantState.scala (hunk 2 above)
Comment (dasormeter) [reply src/main/scala/com/daml/ledger/damlonxexample/ExampleInMemoryParticipantState.scala]: addressed

**Pull request merged**